specs/RainbowBridge/README.md

```diff
@@ -0,0 +1,388 @@
+- The contract `(1)` invokes a method on the `NearTxResultVerifier` `(2)`, passing in the appropriate parameters to represent the transaction result
+- `NearTxResultVerifier` `(2)` unpacks the Near transaction result and extracts the related block header which it passes on to the `NearClient` `(3)`
+- `NearClient` `(3)` returns a boolean indicating whether the block header was indeed recorded on the NEAR blockchain
+- `NearTxResultVerifier` `(2)` passes this result on to the contract `(1)` as proof or rejection of the transaction result as valid
+
+*In both cases, the light clients `EthClient` and `NearClient` are supported by arbitrary trustless actors (`EthRelay` and `NearRelay`) who wish to forward necessary data between Near and Ethereum blockchains, respectively.*
+
+
+### Software
+
+**Core Components**
+
+Two smart contracts, `(C)` and `(3)` establish the baseline source of truth for blockchain activity on the "other" network.
+
+- (C) `NearClient`- *smart contract hosted on **Ethereum** network*  \
+   This contract is a NEAR light client that receives NEAR block headers.  It verifies and stores block hashes only.
+
+- (3) `EthClient` – *smart contract hosted on the **NEAR** network*  \
+   This contract is an Ethereum light client that receives Ethereum block headers.  It verifies ethash and longest chain rule and stores block hashes only.
+
+**Reference Implementations**
+
+Two smart contracts demonstrate what is possible using the Rainbow Bridge.  These contract are intended to be used as **reference material** for developing other contracts about various blockchain platforms. 
+
+*A note about NEAR and Ethereum blockchains*
+
+- At its core, NEAR is a sharded transaction processing engine which records *transaction results*.  These *transaction results* are a verifiable source of truth about the NEAR blockchain.
+- At its core, Ethereum is single threaded computer which, at various points in time, records *events*.  These *events* are a verifiable source of truth about the Ethereum blockchain.
+
+- `NearTxResultVerifier` - *smart contract on **Ethereum** network*   \
+  This contract performs verification of **NEAR transaction results** that have been included into NEAR blocks. It uses Merkle trees and hash preimages for verification.
+
+- `EthEventVerifier` - *smart contract on **NEAR** network*   \
+  This contract performs verification of **Ethereum events** that have been included into Ethereum blocks. It uses Merkle trees and hash preimages for verification.
+
+**Supporting Scripts**
+
+- `NearRelay` - an arbitrary, trustless actor that forwards NEAR block headers to the `NearClient` smart contract which is hosted on the Ethereum network.
+- `EthRelay` - an arbitrary, trustless actor that forwards Ethereum block headers to the `EthClient` smart contract which is hosted on the NEAR network.
+
+
+## Setting up the Rainbow Bridge
+
+1. clone the repo `git clone git@github.com:nearprotocol/near-bridge.git`
```

Comment:
```suggestion
1. clone the repo `git clone --recurse-submodules git@github.com:nearprotocol/near-bridge.git`
```

Reply:
We have Git Submodule in `ethrelay/ethashproof` dir. We also need to call it's `build.sh` script to make EthRelay work.

Reply:
ah, that's where that went!  thank you